pkg/proxy/request/request.go

```diff
@@ -17,3 +17,36 @@
 // Package request provides functionality for handling HTTP Requests
 // including the insertion of configuration options into the request
 package request
+
+import (
+	"context"
+	"net/http"
+
+	tctx "github.com/trickstercache/trickster/v2/pkg/proxy/context"
+)
+
+// Clone wraps the builtin Clone to use a deep clone of both the request body
+// reader (when present) and the Trickster context data
+func Clone(r *http.Request) (*http.Request, error) {
+	if r == nil {
+		return nil, nil
+	}
+	rsc := GetResources(r)
+	if rsc != nil {
+		rsc = rsc.Clone()
+	}
+	ctx := context.Background()
+	if rsc != nil {
+		ctx = tctx.WithResources(ctx, rsc.Clone())
+	}
+	out := r.Clone(context.Background()).WithContext(ctx)
```

Comment:
why not set the context during the initial clone?
```suggestion
	out := r.Clone(ctx)
```